activerecord/lib/active_record/relation/batches.rb

```diff
@@ -122,24 +122,97 @@ def find_in_batches(begin_at: nil, end_at: nil, batch_size: 1000, start: nil)
         end
       end
 
+      in_batches(of: batch_size, begin_at: begin_at, end_at: end_at, load: true) do |batch|
+        yield batch.to_a
+      end
+    end
+
+    # Yields ActiveRecord::Relation objects to work with a batch of records.
+    #
+    #   Person.where("age > 21").in_batches do |relation|
+    #     sleep(50)
+    #     relation.update_all(cool: true)
+    #   end
+    #
+    # If you do not provide a block to #in_batches, it will return an Enumerator
+    # which yields ActiveRecord::Relation objects for chaining with other methods:
+    #
+    #   Person.in_batches.with_index do |relation, batch|
+    #     puts "Processing relation ##{batch}"
+    #     relation.each{|relation| relation.delete_all }
```

Comment:
This should have more spaces: `relation.each { |relation| relation.delete_all }`